maven/bnd-export-maven-plugin/src/main/java/aQute/bnd/maven/export/plugin/ExportMojo.java

```diff
@@ -90,6 +92,12 @@
 	@Parameter(defaultValue = "${session}", readonly = true)
 	private MavenSession				session;
 
+	@Parameter
+	private List<String>				scopes	= Arrays.asList("compile", "runtime");
```

Comment:
Good idea! I'll check that.

Reply:
so trying:
```
@Parameter
private EnumSet<ResolutionScope>	scopes	= EnumSet.of(ResolutionScope.COMPILE, ResolutionScope.RUNTIME);
```
at runtime I get:
```
[INFO] [ERROR] Failed to execute goal biz.aQute.bnd:bnd-export-maven-plugin:4.2.0-SNAPSHOT:export (default) on project export: Execution default of goal biz.aQute.bnd:bnd-export-maven-plugin:4.2.0-SNAPSHOT:export failed: A required class was missing while executing biz.aQute.bnd:bnd-export-maven-plugin:4.2.0-SNAPSHOT:export: org/apache/maven/plugins/annotations/ResolutionScope
```
So seems the annotations are not available at runtime since they are part of the tooling API. I can implement the checking with our own wrapper?

Reply:
OK, I guess Maven did the samething OSGi did with the annotations package by only using CLASS retention annotations and since the enum is in that package, it is not available at runtime.

So the alternate would be for us to create our own enum type:

```java
public enum Scope {
  compile, runtime, ...
}
```

which we use:

```java
@Parameter(defaultValue="compile,scope")
private Set<Scope> scopes;
```

Reply:
It's not that the annotations are CLASS retention, the entire annotations artifact isn't available at runtime. Do we really want to make that a dependency?

Secondly, the @Parameter annotation cannot be used to set multiple values. That's why we set a default map. And I already made an enum type.

Reply:
Well it is not available at runtime since the contents are CLASS retention annotations and supporting enums. Just like org.osgi.service.component.annotations.

Reply:
> the @parameter annotation cannot be used to set multiple values.

OK. So:

```java
@Parameter
private Set<Scope> scopes = EnumSet.of(Scope.compile, Scope.runtime);
```

Reply:
Actually you can't intialize to an enum set since it is not writeable. So you could use a HashSet. But you still cannot populate with values here since there is no way to remove runtime if the config only has compile.

Reply:
We need some test cases for the scope config since the propose initialization appears to be a bug.

Reply:
I still think we need integration tests for the scope configuration.